libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3096,6 +3099,12 @@ bool TypeChecker::visit(Identifier const& _identifier)
 	return false;
 }
 
+void TypeChecker::endVisit(UserDefinedTypeName const& _userDefinedTypeName)
+{
+	if (!_userDefinedTypeName.annotation().type)
+		_userDefinedTypeName.annotation().type = _userDefinedTypeName.pathNode()->annotation().referencedDeclaration->type();
```

Comment:
Where was that done before? And can you change the check into an assertion?

Reply:
Well, before the function that now visits the `IdentifierPath` instead visited the `UserDefinedTypeName`. Here we just copy the reference in the annotation from the IdentifierPath to the UserDefinedTypeName

Reply:
Can you point to the previous implementation? I cannot find it.

Reply:
It seems my explanation was only half-correct, the original assignment was here: https://github.com/ethereum/solidity/blob/develop/libsolidity/analysis/ReferencesResolver.cpp#L174

But now that we removed the UserDefinedTypeName Annotation this part was removed anyway.